# Homicide Rates per County

In [1]:
import pandas as pd 

In [2]:
hom_by_cty = pd.read_csv('cty_homicides_2017.txt', delimiter='\t')
hom_by_cty.head(1).T

0
Notes                       NaN
County       Autauga County, AL
County Code                1001
Deaths               Suppressed
Population                55504
Crude Rate           Suppressed

In [3]:
hom_by_cty.drop(['Notes'], inplace=True, axis=1)

In [4]:
#this is homicide rate per county where there is enough data 
hom_by_cty.head()

County  County Code      Deaths Population  Crude Rate
0  Autauga County, AL       1001.0  Suppressed      55504  Suppressed
1  Baldwin County, AL       1003.0  Suppressed     212628  Suppressed
2  Barbour County, AL       1005.0  Suppressed      25270  Suppressed
3     Bibb County, AL       1007.0  Suppressed      22668  Suppressed
4   Blount County, AL       1009.0  Suppressed      58013  Suppressed

# County Features 

We tried using the ACS 1 year data through Cenpy and the API and consistently got ~820 to 840 counties. 

I tried it from the API directly and from cenpy, with different years for each, and it always returned counties within that range. And, not all states had counties in there, so it wasn't a random sample (remember, Arizona had zero...) 

#### However, using the 5 year data does get us more county information! 

In [5]:
import cenpy as c
#find table that we want to query 
available = c.explorer.available()
acs_df = available[available['title'].str.contains('ACS') == True]
acs_df = acs_df[acs_df['vintage'] == 2017]
acs_df

/Users/tjjj/anaconda3/lib/python3.7/site-packages/libpysal/weights/util.py:19: UserWarning: geopandas not available. Some functionality will be disabled.
  warn('geopandas not available. Some functionality will be disabled.')


variable_cell_limit                                    title  \
ACSDT1Y2017                  NaN               ACS 1-Year Detailed Tables   
ACSCP1Y2017                  NaN           ACS 1-Year Comparison Profiles   
ACSDP1Y2017                  NaN                 ACS 1-Year Data Profiles   
ACSSPP1Y2017                 NaN  ACS 1-Year Selected Population Profiles   
ACSST1Y2017                  NaN                ACS 1-Year Subject Tables   
ACSDT5Y2017                  NaN               ACS 5-Year Detailed Tables   
ACSCP5Y2017                  NaN           ACS 5-Year Comparison Profiles   
ACSDP5Y2017                  NaN                 ACS 5-Year Data Profiles   
ACSST5Y2017                  NaN                ACS 5-Year Subject Tables   
ACSSE2017                    NaN        ACS 1-Year Supplemental Estimates   

                  temporal spatial           publisher programCode  \
ACSDT1Y2017   unidentified     NaN  U.S. Census Bureau     006:004   
ACSCP1Y2017   unidentified     NaN  U.S. Census Bureau     006:004   
ACSDP1Y2017   unidentified     NaN  U.S. Census Bureau     006:004   
ACSSPP1Y2017  unidentified     NaN  U.S. Census Bureau     006:004   
ACSST1Y2017   unidentified     NaN  U.S. Census Bureau     006:004   
ACSDT5Y2017   unidentified     NaN  U.S. Census Bureau     006:004   
ACSCP5Y2017   unidentified     NaN  U.S. Census Bureau     006:004   
ACSDP5Y2017   unidentified     NaN  U.S. Census Bureau     006:004   
ACSST5Y2017   unidentified     NaN  U.S. Census Bureau     006:004   
ACSSE2017     unidentified     NaN  U.S. Census Bureau     006:004   

                           modified keyword  \
ACSDT1Y2017   2018-09-13 00:00:00.0      ()   
ACSCP1Y2017   2018-09-13 00:00:00.0      ()   
ACSDP1Y2017   2018-09-13 00:00:00.0      ()   
ACSSPP1Y2017  2018-09-17 00:00:00.0      ()   
ACSST1Y2017   2018-09-13 00:00:00.0      ()   
ACSDT5Y2017   2018-08-21 07:11:43.0      ()   
ACSCP5Y2017   2018-10-19 00:00:00.0      ()   
ACSDP5Y2017   2018-10-19 00:00:00.0      ()   
ACSST5Y2017   2018-10-19 00:00:00.0      ()   
ACSSE2017     2018-10-18 00:00:00.0      ()   

                                                   distribution  \
ACSDT1Y2017   {'@type': 'dcat:Distribution', 'accessURL': 'h...   
ACSCP1Y2017   {'@type': 'dcat:Distribution', 'accessURL': 'h...   
ACSDP1Y2017   {'@type': 'dcat:Distribution', 'accessURL': 'h...   
ACSSPP1Y2017  {'@type': 'dcat:Distribution', 'accessURL': 'h...   
ACSST1Y2017   {'@type': 'dcat:Distribution', 'accessURL': 'h...   
ACSDT5Y2017   {'@type': 'dcat:Distribution', 'accessURL': 'h...   
ACSCP5Y2017   {'@type': 'dcat:Distribution', 'accessURL': 'h...   
ACSDP5Y2017   {'@type': 'dcat:Distribution', 'accessURL': 'h...   
ACSST5Y2017   {'@type': 'dcat:Distribution', 'accessURL': 'h...   
ACSSE2017     {'@type': 'dcat:Distribution', 'accessURL': 'h...   

                                                    description  ... c_isCube  \
ACSDT1Y2017   The American Community Survey (ACS) is an ongo...  ...     True   
ACSCP1Y2017   The American Community Survey (ACS) is an ongo...  ...     True   
ACSDP1Y2017   The American Community Survey (ACS) is an ongo...  ...     True   
ACSSPP1Y2017  Selected Population Profiles provide broad soc...  ...     True   
ACSST1Y2017   The American Community Survey (ACS) is an ongo...  ...     True   
ACSDT5Y2017   The American Community Survey (ACS) is an ongo...  ...     True   
ACSCP5Y2017   The American Community Survey (ACS) is an ongo...  ...     True   
ACSDP5Y2017   The American Community Survey (ACS) is an ongo...  ...     True   
ACSST5Y2017   The American Community Survey (ACS) is an ongo...  ...     True   
ACSSE2017     The American Community Survey (ACS) is a natio...  ...     True   

              c_isAvailable c_isAggregate  \
ACSDT1Y2017            True          True   
ACSCP1Y2017            True          True   
ACSDP1Y2017            True          True   
ACSSPP1Y2017           True          True   
ACSST1Y2017          

Based on this website and other research, we want to use ACSDP5Y2017.

In [6]:
c.explorer.explain('ACSDP5Y2017')

{'ACS 5-Year Data Profiles': 'The American Community Survey (ACS) is an ongoing survey that provides data every year -- giving communities the current information they need to plan investments and services. The ACS covers a broad range of topics about social, economic, demographic, and housing characteristics of the U.S. population. The data profiles include the following geographies: nation, all states (including DC and Puerto Rico), all metropolitan areas, all congressional districts, all counties, all places and all tracts. Data profiles contain broad social, economic, housing, and demographic information. The data are presented as both counts and percentages. There are over 2,400 variables in this dataset.'}

In [7]:
con = c.base.Connection('ACSDP5Y2017')
g_unit = 'county:*'

In [8]:
#create education features list from https://api.census.gov/data/2017/acs/acs5/profile/variables.html
cols_edu = [] 
for n in range(52,68): 
    var_name = 'DP02_00'+str(n)+'PE'
    cols_edu.append(var_name)

#create internet features list
cols_internet = []
for n in range(150,153):
    var_name = 'DP02_0'+str(n)+'PE'
    cols_internet.append(var_name)

#create row names for joining and EDA
cols_req = ['NAME']


In [9]:
cols = cols_req + cols_edu + cols_internet 
cols

['NAME',
 'DP02_0052PE',
 'DP02_0053PE',
 'DP02_0054PE',
 'DP02_0055PE',
 'DP02_0056PE',
 'DP02_0057PE',
 'DP02_0058PE',
 'DP02_0059PE',
 'DP02_0060PE',
 'DP02_0061PE',
 'DP02_0062PE',
 'DP02_0063PE',
 'DP02_0064PE',
 'DP02_0065PE',
 'DP02_0066PE',
 'DP02_0067PE',
 'DP02_0150PE',
 'DP02_0151PE',
 'DP02_0152PE']

In [10]:
counties_df = con.query(cols=cols, geo_unit=g_unit)
counties_df.head()

NAME DP02_0052PE DP02_0053PE DP02_0054PE DP02_0055PE  \
0    Pickens County, Alabama        4416        11.8         5.7        39.2   
1     Sumter County, Alabama        4106         4.4         5.9        26.9   
2  Jefferson County, Alabama      165739         6.8         5.7        39.9   
3    Choctaw County, Alabama        2718         3.9         4.3        48.0   
4   Franklin County, Alabama        7426         3.3         6.8        49.2   

  DP02_0056PE DP02_0057PE DP02_0058PE DP02_0059PE DP02_0060PE  ...  \
0        26.8        16.5       14241         6.2        13.9  ...   
1        18.8        44.1        8244         4.7        12.6  ...   
2        20.5        27.1      447048         3.0         7.6  ...   
3        24.4        19.4        9449         6.6        13.3  ...   
4        23.3        17.4       20734        11.8        11.9  ...   

  DP02_0063PE DP02_0064PE DP02_0065PE DP02_0066PE DP02_0067PE DP02_0150PE  \
0         7.5         8.8         3.0        79.8        11.8        7620   
1         7.0        10.6         7.6        82.7        18.2        5073   
2         8.1        19.4        12.5        89.4        31.9      261390   
3         9.1         7.9         3.8        80.1        11.6        5463   
4         7.5         8.3         5.1        76.4        13.4       11533   

  DP02_0151PE DP02_0152PE state county  
0        71.0        60.9    01    107  
1        64.8        50.4    01    119  
2        84.4        73.0    01    073  
3        70.4        52.3    01    023  
4        74.2        60.3    01    059  

[5 rows x 22 columns]

In [12]:
counties_df = counties_df.rename(index=str, columns={'NAME': 'county_name', 
                                       'DP02_0052PE': '%_inschool_3+',
                                       'DP02_0053PE': '%_preschool_3+',
                                       'DP02_0054PE': '%_kinderg_3+',
                                       'DP02_0055PE': '%_elementary_3+',
                                       'DP02_0056PE': '%_highschool_3+',
                                       'DP02_0057PE': '%_college_3+',
                                       'DP02_0058PE': '%_25+',
                                       'DP02_0059PE': '%_below9th_25+',
                                       'DP02_0060PE': '%_9th-12th_25+',
                                       'DP02_0061PE': '%_hsgrad_25+',
                                       'DP02_0062PE': '%_somecollege_25+',
                                       'DP02_0063PE': '%_associates_25+',
                                       'DP02_0064PE': '%_bachelors_b25+',
                                       'DP02_0065PE': '%_gradschool_25+',
                                       'DP02_0066PE': '%_hsgrad_or+_25+',
                                       'DP02_0067PE': '%_bachelors_or+_25+',
                                       'DP02_0150PE': '%_useinternet_total_households',
                                       'DP02_0151PE': '%_havecomp_total_households',
                                       'DP02_0152PE': '%_broadband_total_households',
                                      })

In [13]:
counties_df[counties_df['county_name'].str.contains('Wyoming')]

county_name %_inschool_3+ %_preschool_3+ %_kinderg_3+  \
1857       Wyoming County, New York          7920            5.7          4.3   
2267   Wyoming County, Pennsylvania          5697            5.8          4.8   
2966  Wyoming County, West Virginia          4207            6.6          7.6   
3047         Goshen County, Wyoming          2938            5.4          3.2   
3048          Uinta County, Wyoming          5557            6.8          6.3   
3049       Washakie County, Wyoming          1783            4.7          2.5   
3050    Hot Springs County, Wyoming           943           10.0          6.0   
3051        Fremont County, Wyoming         10169            9.1          5.7   
3052       Sublette County, Wyoming          2482            6.8          7.7   
3053         Weston County, Wyoming          1441           13.4          2.7   
3054       Campbell County, Wyoming         12131            5.1          6.2   
3055        Lincoln County, Wyoming          4972            7.2          7.0   
3056        Johnson County, Wyoming          1571            3.9          7.9   
3057        Natrona County, Wyoming         19616            6.2          6.5   
3058        Laramie County, Wyoming         23925            5.5          5.4   
3059       Converse County, Wyoming          3259            9.4          6.2   
3060       Niobrara County, Wyoming           511            2.7          3.1   
3061          Teton County, Wyoming          4752            6.4          2.7   
3062     Sweetwater County, Wyoming         12116            5.9          4.4   
3063         Platte County, Wyoming          1654            4.1          4.7   
3064       Sheridan County, Wyoming          6683            4.8          5.0   
3065       Big Horn County, Wyoming          2933            6.3          4.8   
3066          Crook County, Wyoming          1582            9.2          4.7   
3067           Park County, Wyoming          6337            4.5          6.4   
3068         Albany County, Wyoming         16268            3.3          2.4   
3069         Carbon County, Wyoming          3646           10.4          6.5   

     %_elementary_3+ %_highschool_3+ %_college_3+  %_25+ %_below9th_25+  \
1857            46.4            25.3         18.3  29320            3.5   
2267            42.4            25.4         21.6  19657            1.8   
2966            49.3            24.6         11.9  15859            8.5   
3047            42.1            24.1         25.3   9449            2.0   
3048            53.9            21.6         11.5  12978            2.3   
3049            52.7            33.5          6.7   5705            2.9   
3050            49.9            19.3         14.7   3604            1.2   
3051            45.1            21.1         19.0  26781            2.1   
3052            49.4            21.5         14.6   7051            1.6   
3053            46.2            19.4         18.3   5074            1.5   
3054            52.8            19.6         16.3  30465            2.2   
3055            49.4            22.2         14.3  12453            1.2   
3056            59.0            22.6          6.6   6063            1.5   
3057            42.8            20.0         24.5  54380            1.5   
3058            43.4            19.1         26.6  65206            1.7   
3059            48.7            23.0         12.7   9527            3.4   
3060            30.3            45.2         18.6   1766            3.1   
3061            51.2            13.8         25.8  17008            2.5   
3062            45.7            21.5         22.5  28393            2.1   
3063            52.3            22.4         16.5   6286            2.1   
3064            44.1            22.8         23.3  21072            1.1   
3065            49.1            24.8         15.0   8030            2.8   
3066            54.7            21.9          9.5   5109            2.9   
3067            41.7            21.4

In [14]:
counties_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3220 entries, 0 to 3219
Data columns (total 22 columns):
county_name                       3220 non-null object
%_inschool_3+                     3142 non-null object
%_preschool_3+                    3142 non-null object
%_kinderg_3+                      3142 non-null object
%_elementary_3+                   3142 non-null object
%_highschool_3+                   3142 non-null object
%_college_3+                      3142 non-null object
%_25+                             3142 non-null object
%_below9th_25+                    3142 non-null object
%_9th-12th_25+                    3142 non-null object
%_hsgrad_25+                      3142 non-null object
%_somecollege_25+                 3142 non-null object
%_associates_25+                  3142 non-null object
%_bachelors_b25+                  3142 non-null object
%_gradschool_25+                  3142 non-null object
%_hsgrad_or+_25+                  3142 non-null object
%_bachelors_or+_25

In [15]:
counties_df['county_code'] = counties_df['state'] + counties_df['county']

counties_df.head()

county_name %_inschool_3+ %_preschool_3+ %_kinderg_3+  \
0    Pickens County, Alabama          4416           11.8          5.7   
1     Sumter County, Alabama          4106            4.4          5.9   
2  Jefferson County, Alabama        165739            6.8          5.7   
3    Choctaw County, Alabama          2718            3.9          4.3   
4   Franklin County, Alabama          7426            3.3          6.8   

  %_elementary_3+ %_highschool_3+ %_college_3+   %_25+ %_below9th_25+  \
0            39.2            26.8         16.5   14241            6.2   
1            26.9            18.8         44.1    8244            4.7   
2            39.9            20.5         27.1  447048            3.0   
3            48.0            24.4         19.4    9449            6.6   
4            49.2            23.3         17.4   20734           11.8   

  %_9th-12th_25+  ... %_bachelors_b25+ %_gradschool_25+ %_hsgrad_or+_25+  \
0           13.9  ...              8.8              3.0             79.8   
1           12.6  ...             10.6              7.6             82.7   
2            7.6  ...             19.4             12.5             89.4   
3           13.3  ...              7.9              3.8             80.1   
4           11.9  ...              8.3              5.1             76.4   

  %_bachelors_or+_25+ %_useinternet_total_households  \
0                11.8                           7620   
1                18.2                           5073   
2                31.9                         261390   
3                11.6                           5463   
4                13.4                          11533   

  %_havecomp_total_households %_broadband_total_households state county  \
0                        71.0                         60.9    01    107   
1                        64.8                         50.4    01    119   
2                        84.4                         73.0    01    073   
3                        70.4                         52.3    01    023   
4                        74.2                         60.3    01    059   

  county_code  
0       01107  
1       01119  
2       01073  
3       01023  
4       01059  

[5 rows x 23 columns]

In [16]:
counties_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3220 entries, 0 to 3219
Data columns (total 23 columns):
county_name                       3220 non-null object
%_inschool_3+                     3142 non-null object
%_preschool_3+                    3142 non-null object
%_kinderg_3+                      3142 non-null object
%_elementary_3+                   3142 non-null object
%_highschool_3+                   3142 non-null object
%_college_3+                      3142 non-null object
%_25+                             3142 non-null object
%_below9th_25+                    3142 non-null object
%_9th-12th_25+                    3142 non-null object
%_hsgrad_25+                      3142 non-null object
%_somecollege_25+                 3142 non-null object
%_associates_25+                  3142 non-null object
%_bachelors_b25+                  3142 non-null object
%_gradschool_25+                  3142 non-null object
%_hsgrad_or+_25+                  3142 non-null object
%_bachelors_or+_25

In [17]:
counties_df['county_code'] = counties_df['county_code'].astype(int)
counties_df['state'] = counties_df['state'].astype(int)
counties_df['county'] = counties_df['county'].astype(int)
counties_df.iloc[:,1:-3] = counties_df.iloc[:,1:-3].astype(float)
counties_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3220 entries, 0 to 3219
Data columns (total 23 columns):
county_name                       3220 non-null object
%_inschool_3+                     3142 non-null float64
%_preschool_3+                    3142 non-null float64
%_kinderg_3+                      3142 non-null float64
%_elementary_3+                   3142 non-null float64
%_highschool_3+                   3142 non-null float64
%_college_3+                      3142 non-null float64
%_25+                             3142 non-null float64
%_below9th_25+                    3142 non-null float64
%_9th-12th_25+                    3142 non-null float64
%_hsgrad_25+                      3142 non-null float64
%_somecollege_25+                 3142 non-null float64
%_associates_25+                  3142 non-null float64
%_bachelors_b25+                  3142 non-null float64
%_gradschool_25+                  3142 non-null float64
%_hsgrad_or+_25+                  3142 non-null float64
%_b

In [18]:
counties_df[pd.isnull(counties_df).any(axis=1)]

county_name  %_inschool_3+  %_preschool_3+  \
3142         Jayuya Municipio, Puerto Rico            NaN             NaN   
3143   Quebradillas Municipio, Puerto Rico            NaN             NaN   
3144        Guayama Municipio, Puerto Rico            NaN             NaN   
3145        Guánica Municipio, Puerto Rico            NaN             NaN   
3146         Rincón Municipio, Puerto Rico            NaN             NaN   
3147       Villalba Municipio, Puerto Rico            NaN             NaN   
3148   Aguas Buenas Municipio, Puerto Rico            NaN             NaN   
3149        Bayamón Municipio, Puerto Rico            NaN             NaN   
3150    Hormigueros Municipio, Puerto Rico            NaN             NaN   
3151         Manatí Municipio, Puerto Rico            NaN             NaN   
3152        Naguabo Municipio, Puerto Rico            NaN             NaN   
3153        Humacao Municipio, Puerto Rico            NaN             NaN   
3154     Juana Díaz Municipio, Puerto Rico            NaN             NaN   
3155          Coamo Municipio, Puerto Rico            NaN             NaN   
3156         Dorado Municipio, Puerto Rico            NaN             NaN   
3157           Moca Municipio, Puerto Rico            NaN             NaN   
3158          Lares Municipio, Puerto Rico            NaN             NaN   
3159    Las Piedras Municipio, Puerto Rico            NaN             NaN   
3160     Río Grande Municipio, Puerto Rico            NaN             NaN   
3161       San Juan Municipio, Puerto Rico            NaN             NaN   
3162        Culebra Municipio, Puerto Rico            NaN             NaN   
3163         Gurabo Municipio, Puerto Rico            NaN             NaN   
3164        Salinas Municipio, Puerto Rico            NaN             NaN   
3165    Barceloneta Municipio, Puerto Rico            NaN             NaN   
3166      Vega Alta Municipio, Puerto Rico            NaN             NaN   
3167      Vega Baja Municipio, Puerto Rico            NaN             NaN   
3168          Loíza Municipio, Puerto Rico            NaN             NaN   
3169      Naranjito Municipio, Puerto Rico            NaN             NaN   
3170       Carolina Municipio, Puerto Rico            NaN             NaN   
3171  San Sebastián Municipio, Puerto Rico            NaN             NaN   
...                                    ...            ...             ...   
3190     Guayanilla Municipio, Puerto Rico            NaN             NaN   
3191      Aguadilla Municipio, Puerto Rico            NaN             NaN   
3192      Cabo Rojo Municipio, Puerto Rico            NaN             NaN   
3193        Fajardo Municipio, Puerto Rico            NaN             NaN   
3194         Juncos Municipio, Puerto Rico            NaN             NaN   
3195          Lajas Municipio, Puerto Rico            NaN             NaN   
3196  Sabana Grande Municipio, Puerto Rico            NaN             NaN   
3197     Las Marías Municipio, Puerto Rico            NaN             NaN   
3198       Patillas Municipio, Puerto Rico            NaN             NaN   
3199          Cayey Municipio, Puerto Rico            NaN             NaN   
3200     San Germán Municipio, Puerto Rico            NaN             NaN   
3201         Utuado Municipio, Puerto Rico            NaN             NaN   
3202        Corozal Municipio, Puerto Rico            NaN             NaN   
3203        Maunabo Municipio, Puerto Rico            NaN             NaN   
3204       Peñuelas Municipio, Puerto Rico            NaN             NaN   
3205        Arecibo Municipio, Puerto Rico            NaN             NaN   
3206         Arroyo Municipio, Puerto Rico            NaN             NaN   
3207        Yabucoa Municipio, Puerto Rico            NaN             NaN   
3208         Aguada Municipio, Puerto Rico            NaN             NaN   
3209   Santa Isabel Municipio, Puerto Rico            NaN             NaN   
3210  Trujillo Alto Municip

In [19]:
counties_df = counties_df.dropna()
counties_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3142 entries, 0 to 3141
Data columns (total 23 columns):
county_name                       3142 non-null object
%_inschool_3+                     3142 non-null float64
%_preschool_3+                    3142 non-null float64
%_kinderg_3+                      3142 non-null float64
%_elementary_3+                   3142 non-null float64
%_highschool_3+                   3142 non-null float64
%_college_3+                      3142 non-null float64
%_25+                             3142 non-null float64
%_below9th_25+                    3142 non-null float64
%_9th-12th_25+                    3142 non-null float64
%_hsgrad_25+                      3142 non-null float64
%_somecollege_25+                 3142 non-null float64
%_associates_25+                  3142 non-null float64
%_bachelors_b25+                  3142 non-null float64
%_gradschool_25+                  3142 non-null float64
%_hsgrad_or+_25+                  3142 non-null float64
%_b

In [20]:
counties_df.tail().T

3137  \
county_name                     Rusk County, Wisconsin   
%_inschool_3+                                     2690   
%_preschool_3+                                     6.4   
%_kinderg_3+                                       3.8   
%_elementary_3+                                   53.5   
%_highschool_3+                                   25.1   
%_college_3+                                      11.1   
%_25+                                            10268   
%_below9th_25+                                     4.8   
%_9th-12th_25+                                     8.2   
%_hsgrad_25+                                      42.5   
%_somecollege_25+                                   19   
%_associates_25+                                  10.7   
%_bachelors_b25+                                    10   
%_gradschool_25+                                   4.8   
%_hsgrad_or+_25+                                    87   
%_bachelors_or+_25+                               14.8   
%_useinternet_total_households                    6294   
%_havecomp_total_households                       78.9   
%_broadband_total_households                        66   
state                                               55   
county                                             107   
county_code                                      55107   

                                                   3138  \
county_name                     Vilas County, Wisconsin   
%_inschool_3+                                      3495   
%_preschool_3+                                      6.7   
%_kinderg_3+                                        5.8   
%_elementary_3+                                    46.2   
%_highschool_3+                                    23.1   
%_college_3+                                       18.1   
%_25+                                             16692   
%_below9th_25+                                        1   
%_9th-12th_25+                                      6.4   
%_hsgrad_25+                                         31   
%_somecollege_25+                                    25   
%_associates_25+                                    9.5   
%_bachelors_b25+                                   17.8   
%_gradschool_25+                                    9.3   
%_hsgrad_or+_25+                                   92.6   
%_bachelors_or+_25+                                27.1   
%_useinternet_total_households                    10758   
%_havecomp_total_households                        81.7   
%_broadband_total_households                       68.8   
state                                                55   
county                                              125   
county_code                                       55125   

                                                     3139  \
county_name                     Shawano County, Wisconsin   
%_inschool_3+                                        8769   
%_preschool_3+                                        6.5   
%_kinderg_3+                                          5.7   
%_elementary_3+                                      46.4   
%_highschool_3+                                      24.3   
%_college_3+                                           17   
%_25+                                               29299   
%_below9th_25+                                        2.5   
%_9th-12th_25+                                        6.5   
%_hsgrad_25+                                         43.7   
%_somecollege_25+                                      20   
%_associates_25+                                     11.5   
%_bachelors_b25+                                     10.1   
%_gradschool_25+                                      5.7   
%_hsgrad_or+_25+                                       91   
%_bachelors_or+_25+                                  15.8   
%_useinternet_total_households                      17024   
%_havecomp_total_households                          80.3   
%_broadband_total_ho

P.S. I got the API to return the same information except it wasn't as easy to work with especially because the strings are long and the column names don't autopopulate with the state and county names. Also of note: When I was getting errors in the wrapper they didn't have details, but the errors through the browser version of the API had details.

In [21]:
counties_df = con.query(cols=cols, geo_unit=g_unit)
counties_df.head()

NAME DP02_0052PE DP02_0053PE DP02_0054PE DP02_0055PE  \
0    Pickens County, Alabama        4416        11.8         5.7        39.2   
1     Sumter County, Alabama        4106         4.4         5.9        26.9   
2  Jefferson County, Alabama      165739         6.8         5.7        39.9   
3    Choctaw County, Alabama        2718         3.9         4.3        48.0   
4   Franklin County, Alabama        7426         3.3         6.8        49.2   

  DP02_0056PE DP02_0057PE DP02_0058PE DP02_0059PE DP02_0060PE  ...  \
0        26.8        16.5       14241         6.2        13.9  ...   
1        18.8        44.1        8244         4.7        12.6  ...   
2        20.5        27.1      447048         3.0         7.6  ...   
3        24.4        19.4        9449         6.6        13.3  ...   
4        23.3        17.4       20734        11.8        11.9  ...   

  DP02_0063PE DP02_0064PE DP02_0065PE DP02_0066PE DP02_0067PE DP02_0150PE  \
0         7.5         8.8         3.0        79.8        11.8        7620   
1         7.0        10.6         7.6        82.7        18.2        5073   
2         8.1        19.4        12.5        89.4        31.9      261390   
3         9.1         7.9         3.8        80.1        11.6        5463   
4         7.5         8.3         5.1        76.4        13.4       11533   

  DP02_0151PE DP02_0152PE state county  
0        71.0        60.9    01    107  
1        64.8        50.4    01    119  
2        84.4        73.0    01    073  
3        70.4        52.3    01    023  
4        74.2        60.3    01    059  

[5 rows x 22 columns]

P.S. I got the API to return the same information except it wasn't as easy to work with especially because the strings are long and the column names don't autopopulate with the state and county names. Also of note: When I was getting errors in the wrapper they didn't have details, but the errors through the browser version of the API had details.

# Compare the target to the features


In [22]:
#how many counties are in the target set? 
hom_by_cty.County.nunique()
#note that some are suppressed 

3147

In [23]:
#how many counties are in the target set? 
print(len(counties_df))
#note that these include Puerto Rico

3220


Pretty close! Maybe we can limit by population first (on homicides df) then join the data on the county name. 

In [24]:
import numpy as np
hom_by_cty.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3213 entries, 0 to 3212
Data columns (total 5 columns):
County         3147 non-null object
County Code    3147 non-null float64
Deaths         3148 non-null object
Population     3148 non-null object
Crude Rate     3148 non-null object
dtypes: float64(1), object(4)
memory usage: 125.6+ KB


In [25]:
#convert columns to usable datatypes
hom = hom_by_cty
hom.head()

County  County Code      Deaths Population  Crude Rate
0  Autauga County, AL       1001.0  Suppressed      55504  Suppressed
1  Baldwin County, AL       1003.0  Suppressed     212628  Suppressed
2  Barbour County, AL       1005.0  Suppressed      25270  Suppressed
3     Bibb County, AL       1007.0  Suppressed      22668  Suppressed
4   Blount County, AL       1009.0  Suppressed      58013  Suppressed

In [26]:
len(hom[hom['Crude Rate']!='Suppressed'])

414

Only 414 rows have usable data. We're technically supposed to have 1000 rows, which we start out with, before we remove the ones that are suppressed. What if we looked at suicides instead? 

# Suicide Rates by County
Just out of curiosity, is there more data to work with around suicides?  

In [27]:
suicides = pd.read_csv('suicides.txt', delimiter='\t').drop(['Notes'], axis=1)

print(len(suicides))
suicides.head()
#we don't have all counties here

1084


County  County Code  Deaths  Population  Crude Rate
0    Autauga County, AL       1001.0    12.0     55504.0  Unreliable
1    Baldwin County, AL       1003.0    39.0    212628.0        1.83
2     Blount County, AL       1009.0    11.0     58013.0  Unreliable
3    Calhoun County, AL       1015.0    29.0    114728.0        2.53
4  Covington County, AL       1039.0    12.0     37092.0  Unreliable

In [28]:
suicides.drop(suicides[suicides['Crude Rate'] == 'Unreliable'].index, inplace=True)

In [29]:
print(len(suicides))
suicides.head()

607


County  County Code  Deaths  Population Crude Rate
1     Baldwin County, AL       1003.0    39.0    212628.0       1.83
3     Calhoun County, AL       1015.0    29.0    114728.0       2.53
5     Cullman County, AL       1043.0    21.0     82755.0       2.54
6      DeKalb County, AL       1049.0    20.0     71617.0       2.79
12  Jefferson County, AL       1073.0   109.0    659197.0       1.65

In [30]:
suicides.drop(suicides[suicides['Crude Rate'] == 'Suppressed'].index, inplace=True)
suicides.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 607 entries, 1 to 1083
Data columns (total 5 columns):
County         536 non-null object
County Code    536 non-null float64
Deaths         537 non-null float64
Population     537 non-null float64
Crude Rate     537 non-null object
dtypes: float64(3), object(2)
memory usage: 28.5+ KB


# Homicide Rates 2013-2017 

In [31]:
homicides = pd.read_csv('2013-2017 Homicide Information.txt', delimiter='\t')
homicides = homicides.drop(['Notes'], axis=1)

In [32]:
homicides = homicides.drop(homicides[homicides['Crude Rate']=='Suppressed'].index)

In [33]:
homicides_reliable = homicides.drop(homicides[homicides['Crude Rate']=='Unreliable'].index)
homicides_reliable = homicides_reliable.drop(homicides_reliable[homicides_reliable['Crude Rate'].isna()].index)

In [34]:
homicides_reliable = homicides_reliable.drop(homicides_reliable[homicides_reliable['Deaths']=='Missing'].index)
homicides_reliable = homicides_reliable.drop(homicides_reliable[homicides_reliable['Deaths'].isna()].index)

In [35]:
homicides_reliable['Deaths'].nunique()

207

In [36]:
homicides_reliable.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 633 entries, 1 to 3147
Data columns (total 6 columns):
County               632 non-null object
County Code          632 non-null float64
Deaths               633 non-null object
Population           633 non-null object
Crude Rate           633 non-null object
% of Total Deaths    633 non-null object
dtypes: float64(1), object(5)
memory usage: 34.6+ KB


In [37]:
od = pd.read_csv('2013-2017 Overdose Information.txt', delimiter='\t')
od = od.drop(['Notes'], axis=1)

od = od.drop(od[od['Crude Rate']=='Suppressed'].index)

od_reliable = od_reliable.drop(od_reliable[od_reliable['Crude Rate']=='Unreliable'].index)
od_reliable = od_reliable.drop(od_reliable[od_reliable['Deaths']=='Missing'].index)
od_reliable = od_reliable.drop(od_reliable[od_reliable['Deaths'].isna()].index)
od_reliable = od_reliable.drop(od_reliable[od_reliable['County Code'].isna()].index)
od_reliable.head()

NameError: name 'od_reliable' is not defined

In [ ]:
od_reliable.info()

In [ ]:
od_reliable['County Code'] = od_reliable['County Code'].astype(int)
od_reliable['Deaths'] = od_reliable['Deaths'].astype(int)
od_reliable['Population'] = od_reliable['Population'].astype(int)
od_reliable['Crude Rate'] = od_reliable['Crude Rate'].astype(float)

In [ ]:
od_reliable.info()

In [ ]:
od_reliable['% of Total Deaths'] = od_reliable['% of Total Deaths'].str.replace("%", "")
od_reliable['% of Total Deaths'] = od_reliable['% of Total Deaths'].astype(float)

In [ ]:
od_df = od_reliable

In [ ]:
od_df.head()

In [ ]:
od_df.info()